test/lib/unit_test_service_levels_accessor.hh

```diff
@@ -45,6 +46,11 @@ public:
                 return _sl_controller.invoke_on_all(&service_level_controller::update_service_levels_from_distributed_data);
             });
         }
+
+        virtual ::shared_ptr<service_level_distributed_data_accessor> upgrade_to_v2(cql3::query_processor& qp, service::raft_group0_client& group0_client) const override {
+            return ::static_pointer_cast<service_level_controller::service_level_distributed_data_accessor>(
+                ::make_shared<unit_test_raft_service_levels_accessor>(qp, group0_client, _sl_controller));
```

Comment:
Does each commit really compile? `unit_test_raft_service_levels_accessor` is introduced in a later commit.

Moreover, does this really guarantee that boost tests will use `unit_test_raft_service_levels_accessor`? Did you check that `unit_test_service_levels_accessor::upgrade_to_v2` is really called during tests?

Reply:
moved

And yes, dda is upgraded and the test uses v2 dda